lib/Conversion/XTenToLinalg.cpp

```diff
@@ -70,20 +69,22 @@ static Value applyPad(Location loc, Value input, ArrayRef<int64_t> pad,
       .result();
 }
 
+/// Return a zero-initialized tensor of given size and dtype.
+static Value zeroInit(ArrayRef<int64_t> sizes, mlir::Type elementType, Location loc,
+                         ConversionPatternRewriter &rewriter) {
+  Value initTensor = rewriter.create<linalg::InitTensorOp>(loc, sizes, elementType);
+  Value c0float = rewriter.create<arith::ConstantOp>(
+      loc, FloatAttr::get(elementType, 0.0));
+  return rewriter.create<linalg::FillOp>(loc, c0float, initTensor)
```

Comment:
I prefer `tensor.splat`, since this means the `linalg.init_tensor` can be dropped. However, if this passes establishes an implicit contract that no tensor-to-linalg lowering has to be performed, this would not be allowed.

Reply:
Thanks for the tip. I'll stick with the `linalg.fill` as the result should indeed go through torch-to-linalg lowering.